### PR TITLE
fix qwen2 cache_position for PyTorch Engine when transformers>4.41.2

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

fix https://github.com/InternLM/lmdeploy/issues/1885

Hi @lvhan028 @grimoire @RunningLeon @AllentDan May you help review this pr? Thanks.

## Modification

as titled

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
ref https://github.com/InternLM/lmdeploy/pull/1852

File: lmdeploy/pytorch/models/qwen2.py [hunk 1]
```diff
@@ -133,6 +133,7 @@ def forward(
         past_key_value: Optional[Tuple[torch.Tensor]] = None,
         output_attentions: bool = False,
         use_cache: bool = False,
+        cache_position: Optional[torch.LongTensor] = None,
```
Comment (grimoire) [inline lmdeploy/pytorch/models/qwen2.py:136]: as we will not use these args, just add a `**kwargs` should be ok.

Comment (zhyncs) [reply lmdeploy/pytorch/models/qwen2.py:136]: ok

**Review by grimoire — APPROVED** [review]

**Review by AllentDan — APPROVED** [review]

**Comment by Volta-lemon** [comment]
I had this problem and I saw you solved it. Now do I just need to re-execute pip install lmdeploy[all]==0.4.2?

**Comment by zhyncs** [comment]
@Volta-lemon This PR has been merged and I manually triggered the compilation at https://github.com/zhyncs/lmdeploy-build/actions/runs/9737958121 and https://github.com/zhyncs/lmdeploy-build/actions/runs/9737959536. After the compilation is completed, it can be installed directly.

**Comment by zhyncs** [comment]
@Volta-lemon https://github.com/zhyncs/lmdeploy-build/releases/tag/1f5dd4e
You may add extra args when installing with `pip3` using `--force-reinstall --no-deps`

**Comment by Volta-lemon** [comment]
Thank you for solving my problem

**Pull request merged**